delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViews.scala

```diff
@@ -237,8 +238,29 @@ final class ElasticSearchViews private (
   def fetch(
       id: IdSegment,
       project: ProjectRef
-  ): IO[ElasticSearchViewRejection, ViewResource]        =
-    fetch(id, project, None).map({ case (res, _) => res }).named("fetchElasticSearchView", moduleType)
+  ): IO[ElasticSearchViewRejection, ViewResource]         =
+    fetch(id, project, None).map { case (res, _) => res }.named("fetchElasticSearchView", moduleType)
+
+  /**
+    * Retrieves a current IndexingElasticSearchView resource.
+    *
+    * @param id      the view identifier
+    * @param project the view parent project
+    */
+  def fetchIndexingView(
+      id: IdSegment,
+      project: ProjectRef
+  ): IO[ElasticSearchViewRejection, IndexingViewResource] =
+    fetch(id, project, None)
+      .flatMap { case (res, _) =>
+        res.value match {
+          case v: IndexingElasticSearchView  =>
+            IO.pure(res.as(v))
+          case _: AggregateElasticSearchView =>
+            IO.raiseError(DifferentElasticSearchViewType(defaultViewId, ElasticSearchAggregate, ElasticSearchIndexing))
```

Comment:
I don't understand. `defaultViewId` is the correct variable name

Reply:
the view id is provided as an argument... why is the rejection mentioning the defaultViewId?

Reply:
I've changed it. The rejection says

> Incorrect ElasticsearchView '$id' type: '$provided' provided, expected '$expected'.

Reply:
I'm not sure we're talking about the same thing...
- the fn `fetchIndexingView` takes a `id: IdSegment` argument
- if that view is an AggregateESView, a rejection is returned, however, the rejection uses the `defaultViewId` argument instead of the function argument, which is `id`

Reply:
true, my bad